src/main/java/seedu/command/DeleteCommand.java

```diff
@@ -54,4 +66,20 @@ public boolean equals(Object other) {
         DeleteCommand otherDeleteCommand = (DeleteCommand) other;
         return this.commandStrings.equals(otherDeleteCommand.commandStrings);
     }
+
+
+    protected String prepareDelete() throws AssertionError {
+        String argString = commandStrings.get(0);
+        int delimiterPos = argString.indexOf('/');
+        // the case where delimiterPos = -1 is impossible as
+        // ARGUMENT_FORMAT and ARGUMENT_TRAILING_FORMAT regex requires a '/'
+        assert delimiterPos != -1 : "Each args will need to include minimally a '/' to split arg and value upon";
+        String argType = argString.substring(0, delimiterPos);
+        String argValue = argString.substring(delimiterPos + 1);
+        assert argType.equals("s") : ONLY_SN_ACCEPTED;
+        if (argType.equals("s")) {
+            return argValue;
+        }
+        return "";
```

Comment:
Line 83 is unreachable if you assert that `argType.equals("s")` in line 79. I do think the assertion is a good idea though, so maybe you'd like to just go directly to `return argValue` after line 79?